tests/compiler/test_loops.py

```diff
@@ -57,7 +59,25 @@ def test_for_error_2(snippetcompiler):
         for i in "foo":
         end
     """,
-        "A for loop can only be applied to lists and relations (reported in For(i) ({dir}/main.cf:2))",
+        "A for loop can only be applied to lists and relations. (reported in For(i) ({dir}/main.cf:2))",
```

Comment:
I added the 
`if not isinstance(self.base, Literal):` for the hint because I found it weird here to see : Hint: "foo" is "foo".

Reply:
I like it. One suggestion for further improvement:
- Instead of `not isinstance(self.base, Literal)`, go for `isinstance(self.base, Reference)`. Otherwise I think you'd get the same weird behavior for a literal dict, which is of type `CreateDict` iirc. I think it makes sense to restrict ourselves to variable/attribute references here to keep it concise.
- Even if it is a literal I'd mention the value. Not twice as in your example above, but just once.

Some examples of how I think the output could then look:
- `Hint: 'a.b.c' resolves to 'null'`
- `Hint: `'"foo"' is not a list`
- `Hint: `'{"test": 1}' is not a list`

Just suggestions, wdyt?

Reply:
l like it :)

Reply:
@sanderr I can't rerequest review but it is ready